### PR TITLE
git: ignore C/C++ LSP caches and local configs

### PR DESCRIPTION
**Pull request by x1ddos**
This adds some well-known file and dir names which are not typically checked into the repo.

### CODE REVIEW

**Review by benma — APPROVED** [review]
Hmm normally I'd say that non-project related files belong in your private gitignore, not the project's gitignore, but since I added the gnu global files there too I can't say anything :zipper_mouth_face: :joy:

File: .gitignore [hunk 1]
```diff
@@ -3,6 +3,13 @@
 /releases/temp/
 NOTES.md
```
Comment (benma) [inline .gitignore:4]: remove this too? seems like this is totally misplaced.

Comment (x1ddos) [reply .gitignore]: done

**Comment by x1ddos** [comment]
> Hmm normally I'd say that non-project related files belong in your private gitignore, not the project's gitignore, but since I added the gnu global files there too I can't say anything :zipper_mouth_face: :joy:

:) yeah, true. these files are pretty common though. we even generate compile_commands.json in https://github.com/digitalbitbox/bitbox02-firmware/blob/92686a1/CMakeLists.txt#L59 in every build.

**Pull request merged**